sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -146,17 +146,14 @@ def on_request(self, request):
         :param request: The PipelineRequest object
         :type request: ~azure.core.pipeline.PipelineRequest
         """
-        update_header = False
+        request_id = _Unset()
         if 'request_id' in request.context.options:
             request_id = request.context.options.pop('request_id')
-            update_header = True
         elif self._request_id is not _Unset:
             request_id = self._request_id
-            update_header = True
         elif self._auto_request_id:
             request_id = str(uuid.uuid1())
-            update_header = True
-        if update_header:
+        if not isinstance(request_id, _Unset):
```

Comment:
That's correct, but that's not the usual way of this pattern. Should look more like this:
```python
request_id = unset = _Unset()
....
if request_id is not unset:
   ....
```
you could even not use a class at all
```python
request_id = unset = object()
....
if request_id is not unset:
   ....
```
The official pattern is to create a unique pointer in memory, and because it's a pointer you can use the "is" keyword to compare it's the same object in memory. That's one of the rare scenario when you can check using "is", since you could check that the instance in memory is the same, instead of checking "isinstance" (which is way more expensive than testing a simple pointer comparison).

I put that as a comment, since your code works, just for the discussion.